File: src/config/libs/components/Text.ts
import { getBounds, initializePlugin } from '@/libs/ui';
import { jsonStringify } from '../core';
import { ComponentProps } from './component';

export type Props = ComponentProps & {
};

export function Text({ bounds, track, ...props }: Props) {
    initializePlugin('Text', 'libzic_TextComponent.so');
    // return getComponent('Text', bounds, [{ track }, props]);

    const jsonConfig = jsonStringify({ 
        componentName: 'Text',
        bounds: getBounds(bounds), // TODO rename to getBounds
        ...props,
     });
     console.log('My config:', jsonConfig);

    return [{ COMPONENT: jsonConfig }];
}


// export function getComponent(name: string, position: Position, values: ZicValue[] = []) {
//     return [{ COMPONENT: `${name} ${getPosition(bounds)}` }, values];
// }

File: config/libs/components/Text.lua
local ____lualib = require("lualib_bundle")
local __TS__ObjectAssign = ____lualib.__TS__ObjectAssign
local __TS__ObjectRest = ____lualib.__TS__ObjectRest
local ____exports = {}
local ____ui = require("config.libs.ui")
local getBounds = ____ui.getBounds
local initializePlugin = ____ui.initializePlugin
local ____core = require("config.libs.core")
local jsonStringify = ____core.jsonStringify
function ____exports.Text(____bindingPattern0)
    local props
    local position
    position = ____bindingPattern0.position
    local track = ____bindingPattern0.track
    props = __TS__ObjectRest(____bindingPattern0, {position = true, track = true})
    initializePlugin("Text", "libzic_TextComponent.so")
    local jsonConfig = jsonStringify(__TS__ObjectAssign(
        {
            componentName = "Text",
            bounds = getBounds(position)
        },
        props
    ))
    print("My config:", jsonConfig)
    return {{COMPONENT = jsonConfig}}
end
return ____exports
